### ISSUE TITLE
do your algorithms consolidate the max value from each block?

### ISSUE THREAD

**Issue by elbrandt** [MaxKotlan/Cuda-Find-Max-Using-Parallel-Reduction#1]
Do these algorithms calculate the overall maximum of the data set? or just the maximum in each thread block?

**Comment by MaxKotlan** [comment]
The dataset is spread over multiple thread blocks and then I believe the algorithms calculate the maximum of the entire dataset.
The algorithm uses a unique id (called idx in the code) to reference elements in the dataset which is independent of the blocks and threads.

**Comment by elbrandt** [comment]
I agree sdata[0] will eventually contain the maximum of each thread block.

Pls show me where the maximum from each thread block gets consolidated into
one global maximum.

it looks to me like 99% of your algorithm comes directly from
https://developer.download.nvidia.com/assets/cuda/files/reduction.pdf
and that presentation makes clear in the code comments on the slides that
it is only writing the maximum from each thread block back into global
memory.

E.

**Comment by nicolasbadano** [comment]
I agree with @elbrandt, these algorithms are not returning the global maximum, but only the maximum of the first block.

**Comment by qqwqqw689** [comment]
yees, it is wrong.
